packages/base/src/3dview/mainview.tsx

```diff
@@ -421,6 +439,18 @@ export class MainView extends React.Component<IProps, IStates> {
       this._transformControls.addEventListener('dragging-changed', event => {
         this._controls.enabled = !event.value;
       });
+      // Add event listener for keydown and keyup to enable/disable rotation snapping
+      document.addEventListener('keydown', event => {
+        if (event.key === 'Control') {
+          this._transformControls.rotationSnap = THREE.MathUtils.degToRad(10);
+        }
+      });
+
+      document.addEventListener('keyup', event => {
+        if (event.key === 'Control') {
+          this._transformControls.rotationSnap = null;
+        }
+      });
```

Comment:
we need to remove these event listeners in `componentWillUnmount`

Reply:
Done!

Reply:
I don't see the change related to removing these event listeners

Reply:
Oh sorry, i misinterpreted the comment and moved these listeners to `componentWillUnmount`

Reply:
Removed now

Reply:
I did not mean to remove these handlers, but if you listen to an event on the `document`, you should remove it by calling `removeEventListener` when you don't use it anymore.